pandas/tests/groupby/test_function.py

```diff
@@ -1230,7 +1230,7 @@ def __init__(self, msg="I will raise inside Cython"):
             super().__init__()
             self.msg = msg
 
-        def __eq__(self, other):
+        def __eq__(self, other: object):
```

Comment:
Should it be
```python
def __eq__(self, other: object) -> bool:
```
?

Reply:
the errors were 

```
pandas/core/dtypes/dtypes.py:392: error: "object" has no attribute "categories"
pandas/core/dtypes/dtypes.py:399: error: "object" has no attribute "_ordered"
pandas/core/dtypes/dtypes.py:402: error: "object" has no attribute "_ordered"
pandas/core/dtypes/dtypes.py:403: error: "object" has no attribute "categories"
pandas/core/dtypes/dtypes.py:411: error: "object" has no attribute "categories"
pandas/core/dtypes/dtypes.py:412: error: "object" has no attribute "categories"
pandas/tests/series/test_ufunc.py:286: error: "object" has no attribute "value"
pandas/tests/indexing/test_indexing.py:599: error: "object" has no attribute "value"
```

These failures are not false positives and show that in some cases of comparison an AttributeError would be raised instead of returning False.

the previously referenced SO response suggests using a guard clause. Using anything other than object or Any, mypy gives a similar suggestion...

```
pandas\core\arrays\sparse\dtype.py:93: error: Argument 1 of "__eq__" is incompatible with supertype "object"; supertype defines the argument type as "object"
pandas\core\arrays\sparse\dtype.py:93: note: It is recommended for "__eq__" to work with arbitrary objects, for example:
pandas\core\arrays\sparse\dtype.py:93: note:     def __eq__(self, other: object) -> bool:
pandas\core\arrays\sparse\dtype.py:93: note:         if not isinstance(other, SparseDtype):
pandas\core\arrays\sparse\dtype.py:93: note:             return NotImplemented
pandas\core\arrays\sparse\dtype.py:93: note:         return <logic to compare two SparseDtype instances>```

Reply:
@simonjayhawkins thank you for the comment. I should have marked this as resolved and open a different comment asking about this.

The false positive I was mentioning [here](https://github.com/pandas-dev/pandas/pull/29818#issuecomment-558434385) are referring to the errors I got after I changed the ```other: object``` to ```other: Any```.

Any way, do you suggest to annotate the ```other``` in a different PR? meaning for this PR the only changes are the ```-> bool```.

Reply:
since additional code changes would be required to allow other to be typed correctly as `object`, could do that in a follow on or here.

if not in this PR, can you remove the `Any` and just leave untyped instead (easier to track with `--disallow-incomplete-defs`)

Reply:
> since additional code changes would be required to allow other to be typed correctly as `object`, could do that in a follow on or here.
> 
> if not in this PR, can you remove the `Any` and just leave untyped instead (easier to track with `--disallow-incomplete-defs`)

Thank you for the guidelines @simonjayhawkins